manifests/init.pp

```diff
@@ -121,16 +133,29 @@
   }
 
   if $use_fqdn_real == true {
-    @@host { $::fqdn:
-      ensure       => $fqdn_ensure,
-      host_aliases => $my_fqdn_host_aliases,
-      ip           => $fqdn_ip,
+    if $export_tag == undef {
+      @@host { $::fqdn:
```

Comment:
Aha, so you cannot have a tag on a resource that is undef or empty.

Could you please make a comment above that piece of code that explains this and why the code is duplicated.